src/setinverse.jl

```diff
@@ -0,0 +1,57 @@
+# This file is a part of InverseFunctions.jl, licensed under the MIT License (MIT).
+
+
+"""
+    struct FunctionWithInverse{F,InvF}::Function
+
+A function with an inverse.
+
+Do not construct directly, use [`setinverse(f, invf)`](@ref) instead.
+"""
+struct FunctionWithInverse{F,InvF} <: Function
+    f::F
+    invf::InvF
+end
+
+
+(f::FunctionWithInverse)(x) = f.f(x)
+
+inverse(f::FunctionWithInverse) = FunctionWithInverse(f.invf, f.f)
+
+
+"""
+    setinverse(f, invf)::Function
```

Comment:
Since this can't be guaranteed in general anyway, maybe just
```suggestion
    setinverse(f, invf)
```
?